test/compat/async.test.jsx

```diff
@@ -175,4 +175,46 @@ describe('Async renderToString', () => {
 		const rendered = await promise;
 		expect(rendered).to.equal('<p>ok</p>');
 	});
+
+	it('should render JSX after a urql component', async () => {
```

Comment:
We might also want to try a test where it's triple-y nested as it might warrant a loop similar to the `Array.isArray` one.

Reply:
What do you think this would look like?

If I do something like:

```javascript
		const Context = createContext();

		let c = 0;

		const Fetcher = ({ children }) => {
			c++;
			if (c === 1) {
				throw Promise.resolve();
			}
			return <Fragment>{children}</Fragment>;
		};

		const LazyComponent = lazy(
			async () =>
				function ImportedComponent() {
					return <div>2</div>;
				}
		);

		const NestedLazyComponent = lazy(
			async () =>
				function ImportedComponent() {
					return (
						<Context.Provider>
							<Fetcher>
								<LoadableComponent />
							</Fetcher>
						</Context.Provider>
					);
				}
		);

		const LoadableComponent = ({}) => (
			<Suspense fallback={'...loading'}>
				<LazyComponent />
			</Suspense>
		);

		const NestedLoadableComponent = ({}) => (
			<Suspense fallback={'...loading'}>
				<NestedLazyComponent />
			</Suspense>
		);

		const rendered = await renderToStringAsync(
			<Context.Provider>
				<Fetcher>
					<NestedLoadableComponent />
				</Fetcher>
			</Context.Provider>
		);

		expect(rendered).to.equal(`<div>2</div>`);
```

Then the current solution continues to work